db/migrate/20240207191223_remove_entitlement_mode_host_statuses.rb

```diff
@@ -0,0 +1,25 @@
+class RemoveEntitlementModeHostStatuses < ActiveRecord::Migration[6.1]
+  def up
+    obsolete_statuses = [
+      "Katello::SubscriptionStatus",
+      "Katello::PurposeStatus",
+      "Katello::PurposeAddonsStatus",
+      "Katello::PurposeRoleStatus",
+      "Katello::PurposeSlaStatus",
+      "Katello::PurposeUsageStatus"
+    ]
+
+    ::HostStatus::Status.where(type: obsolete_statuses).delete_all
+
+    ::TablePreference.where(name: "hosts").each do |table_preference|
```

Comment:
I think Fake Class for this will be a better idea. If table preference class gets renamed etc in the future we 'll have to revisit.